packages/back-end/src/scim/users/patchUser.ts

```diff
@@ -0,0 +1,148 @@
+import { Response } from "express";
+import { cloneDeep } from "lodash";
+import { updateOrganization } from "../../models/OrganizationModel";
+import {
+  getUserByExternalId,
+  getUserById,
+  removeExternalId,
+} from "../../services/users";
+import { ScimUpdateRequest } from "../../../types/scim";
+import { OrganizationInterface } from "../../../types/organization";
+import { UserInterface } from "../../../types/user";
+
+type Operation = {
+  op: "add" | "remove" | "replace";
+  path: string; // Path is optional for add & replace, and required for remove operations
+  value: {
+    [key: string]: unknown;
+  };
+};
+
+type ScimEmailObject = {
+  primary: boolean;
+  value: string;
+  type: string;
+  display: string;
+};
+
+type ScimUserObject = {
+  schemas: string[];
+  id: string;
+  userName: string;
+  name: {
+    displayName: string;
+  };
+  active: boolean;
+  emails: ScimEmailObject[];
+  groups: string[];
+  meta: {
+    resourceType: string;
+  };
+};
+
+async function removeUserFromOrg(
+  org: OrganizationInterface,
+  userIndex: number,
+  user: UserInterface,
+  updatedScimUser: ScimUserObject
+) {
+  const updatedOrg = cloneDeep(org);
+
+  // TODO: When we introduce the ability to manage roles via SCIM, we can remove this check.
+  const userIsAdmin = org.members[userIndex].role === "admin";
+
+  if (userIsAdmin) {
+    const numberOfAdmins = org.members.filter(
+      (member) => member.role === "admin"
+    );
+
+    if (numberOfAdmins.length === 1) {
+      throw new Error("Cannot remove the only admin");
+    }
+  }
+
+  updatedOrg.members.splice(userIndex, 1);
+
+  await updateOrganization(org.id, updatedOrg);
```

Comment:
With this, the `externalId` will always persist when we deactivate a user. WIth Okta, this could lead to multiple users in our database that share the same `externalId` as if an email address/username is updated within Okta, Okta will make a PATCH request to deactivate the current userName, before making a POST request to create a new user with the new userName.

This will result in two users in our DB that share an `externalId`.  I can't think of any bugs this would introduce, but it smells funky to me.

Reply:
I think this should be fine since we'll be relying on the `managedByIdp` boolean now instead of the `externalId`. I'll try this exact flow and see what happens